src/python/pants/engine/internals/parametrize.py

```diff
@@ -40,11 +40,26 @@ class Parametrize:
 
     args: tuple[str, ...]
     kwargs: FrozenDict[str, ImmutableValue]
+    is_group: bool
 
     def __init__(self, *args: str, **kwargs: Any) -> None:
+        object.__setattr__(self, "is_group", False)
         object.__setattr__(self, "args", args)
         object.__setattr__(self, "kwargs", FrozenDict.deep_freeze(kwargs))
 
+    def keys(self) -> tuple[str]:
+        return (f"parametrize_{hash(self)}",)
```

Comment:
The group name is part of the address, so, it probably needs to be unique. Is there anything that prevents reusing the group name? Or could I do
```python
target(
    **parametrize("A", a=1),
    **parametrize("A", a=2),
)
```

I wonder if we should use something like (though we would need to sanitize it to have a valid python var name. Eg swap `-` for `_`).

```suggestion
        return (f"__parametrize_{self.group_name}__",)
```

That will mean parsing the BUILD file will die if someone reuses a group name, though I'm not sure how useful the error message might be.

I also wonder how that would interact with `__defaults__`. If a group is defined in defaults and on a target, would that error, override the parametrize group, or somehow extend the group?

Reply:
That's an excellent idea, not because it will error out, but because it will override the previous one, so you get only one.. the duplicated entries are simply replaced :)

Reply:
Like so:
https://github.com/pantsbuild/pants/pull/20065/files#diff-46e50cf77c151e5324c9b847605099e9adc992472d4bf1903fd9c8bff27879e0R115-R122